### PR TITLE
More API field updates.

### PR DESCRIPTION
**Pull request by jd28**
I admit some of these are not super high value.

### CODE REVIEW

File: api/CNWItem.h [hunk 1]
```diff
@@ -15,8 +15,8 @@ class CNWItem
     CNWItem();
 
 public:
-    /* 0x0/0 */ unsigned long field_0;
-    /* 0x4/4 */ unsigned long field_4;
+    /* 0x0/0 */ unsigned char LayeredTextureColor[6];
+    /* 0x6/6 */ unsigned short field_4;
```
Comment (virusman) [inline api/CNWItem.h]: field_6
sorry for nitpicking :smiling_imp:

File: api/CNWSItem.h [hunk 2]
```diff
@@ -114,29 +114,60 @@ class CNWSItem : public CNWItem, public CNWSObject
...
     /* 0x244/580 */ unsigned long Possessor;
-    /* 0x248/584 */ unsigned long field_248;
-    /* 0x24C/588 */ char Repos_PosX;
-    /* 0x24D/589 */ char Repos_PosY;
-    /* 0x250/592 */ char rsvd2[2];
-    /* 0x250/592 */ unsigned long field_250;
-    /* 0x288/648 */ char rsvd3[52];
-    /* 0x288/648 */ unsigned long Dropable;
+    /* 0x248/584 */ unsigned long RepositoryPtr;
+    /* 0x24C/588 */ unsigned long field_24C;
```
Comment (virusman) [inline api/CNWSItem.h]: Repos_PosX & Repos_PosY have disappeared

**Comment by jd28** [comment]
Oops.  I have updated,

File: api/CNWRules.h [hunk 3]
```diff
@@ -53,5 +54,42 @@ class CNWRules
...
+    /* 0xF4/244 */ unsigned long Difficulty_1;
+    /* 0xF8/248 */ unsigned long Difficulty_2;
+    /* 0xFC/252 */ unsigned long Difficulty_3;
+    /* 0x100/256 */ unsigned long Difficulty_4;
+    /* 0x104/260 */ unsigned long Difficulty_5;
+    /* 0x108/264 */ unsigned long Difficulty_6;
+    /* 0x10C/268 */ unsigned long Difficulty_7;
+    /* 0x110/272 */ unsigned long Difficulty_8;
+    /* 0x114/276 */ unsigned long Difficulty_9;
+    /* 0x118/280 */ void *vtable;
```
Comment (virusman) [inline api/CNWRules.h]: vptr

**Comment by virusman** [comment]
Thanks!

**Pull request merged**